packages/skill-lesson/hooks/use-mux-player.tsx

```diff
@@ -71,14 +70,24 @@ export const VideoProvider: React.FC<
   // and we need to be able to robustly check for access
   // while understanding what the actual thing
   // being displayed **is**
-  const {data: abilityRules, status: abilityRulesStatus} =
-    trpc.workshops.verifyAccess.useQuery({
-      moduleSlug: module.slug.current,
-      moduleType: module.moduleType,
-      lessonSlug: exerciseSlug,
-      sectionSlug: section?.slug,
-      isSolution: lesson._type === 'solution',
-    })
+  const {data: abilityRules, status: abilityRulesStatus} = useQuery(
+    ['ability-rules', module.slug.current, exerciseSlug],
+    async () => {
+      return await fetch(`/api/skill/modules/verify-access`, {
+        method: 'POST',
+        body: JSON.stringify({
+          moduleSlug: module.slug.current,
+          moduleType: module.moduleType,
+          lessonSlug: exerciseSlug,
+          sectionSlug: section?.slug,
+          isSolution: lesson._type === 'solution',
+        }),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }).then((response) => response.json())
+    },
```

Comment:
todo: base ability on module access instead of this full stack

the ability needs all this info because we allow the first lesson of a module/section to be viewed freely, but that requires a ton of information.